src/chainstate/coordinator/mod.rs

```diff
@@ -556,17 +622,576 @@ impl<
+        )
+        .expect("BUG: failed to query sortition DB")
+        {
+            // it exists on this sortition history, but do we have it in the chainstate?
+            let present = StacksChainState::has_stacks_block(
+                &self.chain_state_db.db(),
+                &StacksBlockHeader::make_index_block_hash(
+                    &sn.consensus_hash,
+                    &block_commit.block_header_hash,
+                ),
+            )
+            .expect("BUG: failed to query chainstate DB");
+            if present {
+                test_debug!(
+                    "Have processed unaffirmed PoX anchor block {}/{} (burn height {})",
+                    &sn.consensus_hash,
+                    &block_commit.block_header_hash,
+                    sn.block_height
+                );
+                present
+            } else {
+                // have we instead maybe downloaded it but not processed it yet?
+                // NOTE: if the anchor block is unprocessable, it will eventually get orphaned
+                test_debug!(
+                    "Have NOT processed unaffirmed PoX anchor block {}/{} (burn height {})",
+                    &sn.consensus_hash,
+                    &block_commit.block_header_hash,
+                    sn.block_height
+                );
+                let has_staging = StacksChainState::has_staging_block(
+                    &self.chain_state_db.db(),
+                    &sn.consensus_hash,
+                    &block_commit.block_header_hash,
+                )
+                .expect("BUG: failed to query chainstate DB");
+                if has_staging {
+                    test_debug!(
+                        "Have unprocessed staging PoX anchor block {}/{} (burn height {})",
+                        &sn.consensus_hash,
+                        &block_commit.block_header_hash,
+                        sn.block_height
+                    );
+                    true
+                } else {
+                    test_debug!(
+                        "Do NOT have unprocessed staging PoX anchor block {}/{} (burn height {})",
+                        &sn.consensus_hash,
+                        &block_commit.block_header_hash,
+                        sn.block_height
+                    );
+                    false
+                }
+            }
+        } else {
+            test_debug!(
+                "No block snapshot for PoX anchor block {} off of sortition {}",
+                &block_commit.block_header_hash,
+                &tip.sortition_id
+            );
+            return false;
+        }
+    }
+
+    pub fn get_canonical_affirmation_map(&self) -> Result<AffirmationMap, Error> {
+        // if we don't have an unaffirmed anchor block, and we're no longer in the initial block
+        // download, then assume that it's absent.  Otherwise, if we are in the initial block
+        // download but we don't have it yet, assume that it's present.
```

Comment:
What is this comment referencing?

Reply:
Oh, I see -- it's referring the anonymous function arg. Maybe move this comment to that check?

Reply:
Sure, moved